### ISSUE TITLE
ipv6 forwarded headers cause NumberFormatException in ConnectionInfo

### ISSUE THREAD

**Issue by RayJSeth** [reactor/reactor-netty#625]
When netty handles a request containing an x-forwarded-* header with ipv6 format, no port, and no brackets, the last segment is taken as port and Integer.parse throws NumberFormatException on the hex or compressed 0. 

parseAddress should infer when ipv6 but no brackets to prevent parsing last segment.

PR [here](https://github.com/reactor/reactor-netty/pull/626)

### Steps to reproduce

Send request to reactor netty with 'x-forwarded-for: 1fff:0:a88:85a3::ac1f', this should result in NumberFormatException -> ConnectionInfo.java:121

### Reactor Netty version
v0.8.5

### JVM version (e.g. `java -version`)
1.8.0_201-b09

**Comment by violetagg** [comment]
Fixed by #626

**Comment by fruedaCode** [comment]
If the header value is this one: [2001:db8:a0b:12f0::1]:dba2 it gives a NumberFormatException.
Is there any way to ignore it or any fallback method to get the port?

**Comment by violetagg** [comment]
@fruedaCode Do you have `reactor.netty.http.server.HttpServer#forwarded` enabled?

**Comment by fruedaCode** [comment]
I'm using spring boot app and this flag is true:
server.use-forward-headers

**Comment by violetagg** [comment]
@fruedaCode please try 0.8.9.RELEASE

**Comment by fruedaCode** [comment]
The fix is not completely right.
Because if the pattern does not match, your're creating  and returning this at the end of 
```
parseAddress(String address, int defaultPort):
```
```
return InetSocketAddressUtil.createUnresolved(address, defaultPort);
```
and the address still contains the port....

So the InetSocketAddress is created wrongly and is throwing a NullPointerException later....

You should remove the port part from address parameter.

**Comment by violetagg** [comment]
@fruedaCode fixed with #770
